mlrun/datastore/model_provider/huggingface_provider.py

```diff
@@ -224,7 +227,8 @@ def load_client(self) -> None:
 
             self.options["model_kwargs"] = self.options.get("model_kwargs", {})
             self.options["model_kwargs"]["local_files_only"] = True
-            self._client = pipeline(model=self.model, **self.options)
+            with self._client_lock:
+                self._client = pipeline(model=self.model, **self.options)
```

Comment:
What's the purpose of this lock?

Reply:
The reason is that the pipeline initialization is not thread safe.
But I think this might not cover all the cases as you (@tomerm-iguazio) already saw... so maybe we should raise an error in case the user wants to use the same HF model from diffrent threads.. and to advise to use the process pool mechanism instead.

Reply:
Can you please add a comment on the jira what is still failing with this fix?
Also, if we end up thinking that we need to advise the user tha he should use the process mechanism when reusing the same HF model, maybe we should not skip this locking?